### DISCUSSION TITLE
[help] Combining non-interactive and interactive workflow

### DISCUSSION THREAD

**Discussion by mdmanurung** [ropensci/targets#1203] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

First of all, thank you for such an awesome package.

I want to ask about how to create targets in both interactive and non-interactive manner. In my data analysis work, there are parts that I can do without much interventions (e.g. aligning reads). However, there are also parts that need a lot of interactive exploration (e.g. quality control). For the former, I can put the sequence of targets in the `_targets.R`. For the latter, however, I can't succeed in adding a target markdown chunk in my analysis rmarkdown file. 

From what I can understand from the tutorial, I can use the pre-generated targets from the non-interactive pipeline for further exploration in the report. However, it is not clear to me how I can make a new targets WITHIN the report. I tried the target markdown but it seems that it serves as a literate alternative for writing _targets.R file but not as a complement?

Thank you for taking the time.

Regards,
Mikhael

**Comment by wlandau** [comment, marked as answer]
> I tried the target markdown but it seems that it serves as a literate alternative for writing _targets.R file but not as a complement?

Yes, target markdown is just an interface to write `_targets.R`.

With the exception of some forms of [debugging](https://books.ropensci.org/targets/debugging.html), interactive tasks are best done outside the pipeline. You can access output with `tar_read()` or `tar_load()`, then run interactive code to do ad hoc testing, either in your interactive R session or a literate programming report that you keep separate from the pipeline. If you find yourself needing to manually interact with a running target, it may be time to choose an alternative approach to organizing that part of your project.

**Reply by mdmanurung** [reply]
Thank you for the clarification!
